File: remove_element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        index = 0

        for i in range (len(nums)):
            if nums[i] != val:
                nums[index] = nums[i]
                index+=1

        return index

        
solution = Solution()
print(solution.removeElement([3,2,2,3],3))
print(solution.removeElement([0,1,2,2,3,0,4,2],2))

print(solution.removeElement([2],2))

File: remove_element.java
public class remove_element {

    public int removeElement(int[] nums, int val) {
        int tracker = 0;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[tracker] = nums[i];
                tracker++;
            }
        }
        return tracker;
    }

    
    public void main(String[] args) {
        removeElement(new int[]{3,2,2,3},3);
        removeElement(new int[]{0,1,2,2,3,0,4,2},2);
    }
}
